### ISSUE TITLE
Add Instance Methods to TensorFlow Frontend

### ISSUE THREAD

**Issue by jkeane508** [unifyai/ivy#3616]
Add [Instance Methods](https://www.tensorflow.org/api_docs/python/tf/Tensor) to TensorFlow Frontend:

- [ ] eval
- [ ] get_shape
- [ ] ref
- [ ] set_shape
- [ ] __abs__
- [ ] __add__
- [ ] __and__
- [ ] __array__
- [ ] __bool__
- [ ] __div__
- [ ] __eq__
- [ ] __floordiv__
- [ ] __ge__
- [ ] __getitem__
- [ ] __gt__
- [ ] __invert__
- [ ] __iter__
- [ ] __le__
- [ ] __len__
- [ ] __lt__
- [ ] __matmul__
- [ ] __mod__
- [ ] __mul__
- [ ] __ne__
- [ ] __neg__
- [ ] __nonzero__
- [ ] __or__
- [ ] __pow__
- [ ] __radd__
- [ ] __rand__
- [ ] __rdiv__
- [ ] __rfloordiv__
- [ ] __rmatmul__
- [ ] __rmod__
- [ ] __rmul__
- [ ] __ror__
- [ ] __rpow__
- [ ] __rsub__
- [ ] __rtruediv__
- [ ] __rxor__
- [ ] __sub__
- [ ] __truediv__
- [ ] __xor__

**Comment by Aditya-Komaravolu** [comment]
@jkeane508 should we add these functions to [ivy/functional/frontends/tensorflow/tensor.py](https://github.com/unifyai/ivy/blob/master/ivy/functional/frontends/tensorflow/tensor.py)

**Comment by emmyidehen** [comment]
#4644

**Comment by jkeane508** [comment]
> @jkeane508 should we add these functions to [ivy/functional/frontends/tensorflow/tensor.py](https://github.com/unifyai/ivy/blob/master/ivy/functional/frontends/tensorflow/tensor.py)?

Yep! This is the frontend class where we should add these instance methods :)

**Comment by Aditya-Komaravolu** [comment]
> > @jkeane508 should we add these functions to [ivy/functional/frontends/tensorflow/tensor.py](https://github.com/unifyai/ivy/blob/master/ivy/functional/frontends/tensorflow/tensor.py)?
> 
> Yep! This is the frontend class where we should add these instance methods :)

Okay. But `abs` is from `tf.math` (as per documentation), so shouldn't we add abs to math.py in the tf frontend?

**Comment by jkeane508** [comment]
> > > @jkeane508 should we add these functions to [ivy/functional/frontends/tensorflow/tensor.py](https://github.com/unifyai/ivy/blob/master/ivy/functional/frontends/tensorflow/tensor.py)?
> > 
> > 
> > Yep! This is the frontend class where we should add these instance methods :)
> 
> Okay. But `abs` is from `tf.math` (as per documentation), so shouldn't we add abs to math.py in the tf frontend?

Yes you are right, but here we are implementing the Instance Methods for the specific frontend array class, documented [here](https://lets-unify.ai/ivy/deep_dive/16_ivy_frontends.html#classes-and-instance-methods)

**Comment by Aditya-Komaravolu** [comment]
> Instance Methods for the specific frontend array class, documented [here](https://lets-unify.ai/ivy/deep_dive/16_ivy_frontends.html#classes-and-instance-methods)

Okay, I get that now. So do I have to move abs to here? Or else, should I leave it as it is in math.py?

**Comment by jkeane508** [comment]
> 

Yep, once you have ensured that the function is implemented in the relevant frontend you would then use that as part of your instance method implementation as per the examples on the docs referenced above :)

**Comment by Aditya-Komaravolu** [comment]
Okay then, I am creating a new commit to add these changes to my existing pull request #4663.

**Comment by khimraj** [comment]
- [ ] #4833

**Comment by khimraj** [comment]
- [ ] #5186

**Comment by khimraj** [comment]
- [ ] #5187

**Comment by Abheast12** [comment]
- [ ] #5307

**Comment by AbhishekJha22** [comment]
__pow__ #5480

**Comment by khimraj** [comment]
- [ ] #5577

**Comment by khimraj** [comment]
- [ ] #5578

**Comment by khimraj** [comment]
- [ ] #5579

**Comment by khimraj** [comment]
- [ ] #5580

**Comment by khimraj** [comment]
- [ ] #5581

**Comment by khimraj** [comment]
- [ ] #5582

**Comment by valeriocost** [comment]
[] #5667

**Comment by kurshakuz** [comment]
- [ ] #5694

**Comment by StefanHUB** [comment]
#7073

**Comment by Taciturny** [comment]
- [ ] #7097

**Comment by nadidebeyza** [comment]
- [ ]  __iter__ #7123

**Comment by hossamEldeinMohmed** [comment]
#9563

**Comment by hossamEldeinMohmed** [comment]
#9564

**Comment by gretagh93** [comment]
- [ ] #11837

**Comment by akhand769** [comment]
#12084

**Comment by akhand769** [comment]
@jkeane508 this pow function is to be implemented similar to the pow function in tensorflow i.e. tf.math.pow ?

**Comment by AkshayC2001** [comment]
- [ ] #13041

**Comment by Sarath2206** [comment]
#16174

**Comment by rullo16** [comment]
eval - #16568

**Comment by ligia26** [comment]
[__add__] #18054

**Comment by Rashed6122** [comment]
ref #18187

**Comment by Rashed6122** [comment]
ref #18187

**Comment by mtilerisoy** [comment]
- [ ] #22223

**Comment by hainesdata** [comment]
- [] #22674

**Comment by musthofasyarifudin** [comment]
#23135

**Comment by ShruAgarwal** [comment]
- [ ] #26140

**Comment by Dheeraj-Bal** [comment]
#26463

**Comment by BirsenYY** [comment]
- eval #26869

**Comment by BirsenYY** [comment]
@jkeane508 
I  checked the method 'eval' from tensorflow documentation which states that this function is not called in 'eager execution'.  I'm supposed to add 'eval' under the class 'EagerTensor'. Does the name of the tensor class suggest that the tensor is built with eager execution which does not involve construction of graphs? 
when writing 'self.' the method 'eval' appears. Does this method automatically take care of eager execution settings?
Can you guide me please? Thanks.

**Comment by Darshan-H-E** [comment]
- [ ] #27055

**Comment by Darshan-H-E** [comment]
- [ ] #27056

**Comment by Darshan-H-E** [comment]
- [ ] #27057

**Comment by Darshan-H-E** [comment]
- [ ] #27058
